crepecake/src/test/java/com/google/cloud/tools/crepecake/image/LayerBuilderTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.image;
+
+import com.google.cloud.tools.crepecake.blob.Blob;
+import com.google.cloud.tools.crepecake.blob.Blobs;
+import com.google.cloud.tools.crepecake.tar.TarStreamBuilder;
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/** Tests for {@link LayerBuilder}. */
+@RunWith(MockitoJUnitRunner.class)
+public class LayerBuilderTest {
+
+  @Mock private TarStreamBuilder mockTarStreamBuilder;
+
+  @InjectMocks private LayerBuilder layerBuilder;
```

Comment:
Interesting. I haven't seen this technique before. 

That said, I'm still not sure you have to mock this. Could we use real files instead of fake files and then verify the real tar that's built? That would be a more comprehensive test.